src/lib/comms/sol-network-impl-linux.c

```diff
@@ -532,7 +535,7 @@ sol_network_link_set_status(uint16_t link_index, unsigned int changes, unsigned
     uint8_t mode = IN6_ADDR_GEN_MODE_EUI64;
     size_t buf_size = sizeof(buf);;
 
-    if (sol_network_start_netlink() < 0) return false;
+    if (sol_network_start_netlink() < 0) return -ENOMEM;
```

Comment:
No no, you can't simple return `-ENOMEM` here. The function `sol_network_start_netlink()` already returns the error value. You should do the following:

```
int r;
...
r = sol_network_start_netlink();
SOL_INT_CHECK(r, < 0, r);
```